tests/integration/runners/test_state.py

```diff
@@ -382,3 +382,88 @@ def test_parallel_orchestrations(self):
+        be pulling from the pillar cache that all the state files are available and
+        the file_roots has been preserved.  See issues #48277 and #46986.
+        '''
+        self.write_conf({
+            'fileserver_backend': ['roots'],
+            'file_roots': {
+                'base': [self.base_env],
+            },
+        })
+
+        orch_sls = os.path.join(self.base_env, 'main.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                include:
+                  - one
+                  - two
+                  - three
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'one.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                {%- set foo = salt['saltutil.runner']('pillar.show_pillar') %}
+                placeholder_one:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'two.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                placeholder_two:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'three.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                placeholder_three:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'main.sls')
+
+        listener = salt.utils.event.get_event(
+            'master',
+            sock_dir=self.master_opts['sock_dir'],
+            transport=self.master_opts['transport'],
+            opts=self.master_opts)
+
+        start_time = time.time()
+        jid = self.run_run_plus(
+            'state.orchestrate',
+            'main',
+            __reload_config=True).get('jid')
+
+        if jid is None:
+            raise Exception('jid missing from run_run_plus output')
+
+        signal.signal(signal.SIGALRM, self.alarm_handler)
+        signal.alarm(self.timeout)
+        received = False
+        try:
+            while True:
+                event = listener.get_event(full=True)
+                if event is None:
+                    continue
+                if event['tag'] == 'salt/run/{0}/ret'.format(jid):
+                    received = True
+                    # Don't wrap this in a try/except. We want to know if the
+                    # data structure is different from what we expect!
+                    ret = event['data']['return']['data']['master']
+                    for state in ret:
+                        data = ret[state]
+                        # Each state should be successful
+                        self.assertEqual(data['comment'], 'Success!')
+                    break
+        finally:
+            self.assertTrue(received)
+            del listener
```

Comment:
BTW, is this really needed?

Reply:
Which part specifically?  The assertTrue or the clean up of the listener?

Reply:
The `del`. Seems you're basically calling `SaltEvent.__del__` in order to call `SaltEvent.destroy`. I would say, we should not do this, because `__del__` is not an opposite to `__init__`. I would rather refactor it a bit:

```python
class SaltEvent(event):
    def destroy(self):
        try:
            if self.subscriber is not None:
                self.subscriber.close()
            if self.pusher is not None:
                self.pusher.close()
            if self._run_io_loop_sync and not self.keep_loop:
                self.io_loop.close()
        except Exception as err:
            log.error('Unable to close event listener')
            log.debug(err)

    __del__ = destroy
```

And then we should never use it as `del obj` but always `obj.destroy()` instead. However, we still leave `__del__` to the garbage collector.